code/game/objects/items/chainsaw.dm

```diff
@@ -78,6 +78,40 @@
 		var/datum/action/A = X
 		A.UpdateButtonIcon()
 
+/obj/item/chainsaw/afterattack(atom/A, mob/user, proximity)
+	if(!on)
+		return ..()
+
+	var/lignocellulose = counterlist_sum(A.has_material_type(/datum/material/wood))
+
+	if(!lignocellulose)
+		return ..()
+
+
+
+	var/dust_harvest = lignocellulose * 0.1 //you lose 10% as sawdust
+	lignocellulose *= 0.9
+	dust_harvest = round((lignocellulose % MINERAL_MATERIAL_AMOUNT) / 100) //the leftover wood material also becomes sawdust.
+	var/plank_harvest = round(lignocellulose / (MINERAL_MATERIAL_AMOUNT))
+
+	if(plank_harvest >= 3)
+		user.visible_message("<span class='notice'>[user] starts sawing [A] to pieces!</span>", "<span class='notice'>You start sawing [A] to pieces!</span>")
+		if(!do_after(user, 2 SECONDS, target = A))
+			return
+
+	if(plank_harvest)
+		new /obj/item/stack/sheet/mineral/wood(get_turf(A),plank_harvest) //spawn an amount of planks equal to plank_harvest
+
+	if(lignocellulose % MINERAL_MATERIAL_AMOUNT)
+		new /obj/effect/decal/cleanable/sawdust(get_turf(A), dust_harvest) //spawn an amount of cellulose inside the decal equal to dust_harvest.
+	playsound(src, 'sound/weapons/chainsawhit.ogg', 50, TRUE)
+	to_chat(user, "<span class='notice'>You saw [A] into [plank_harvest ? "planks" : "splints"].</span>")
+	if(isturf(A))
+		var/turf/scrape_target = A
+		scrape_target.ScrapeAway(flags = CHANGETURF_INHERIT_AIR)
+	else
+		qdel(A)
```

Comment:
`deconstruct`?

Reply:
"A" might be an atom that doesn't have deconstruct such as a mob. 

And many wooden structures drop planks on deconstruct which would lead to material duping.